### PR TITLE
fix: curl import curly bracket fix

### LINKED ISSUE #19846: [Bug]: cUrl import fails when url contains curly brackets around variable names
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When creating a new API from cUrl import, the import fails if there are curly brackets in the url. 

Many REST APIs have a record ID in the endpoint path, and the docs will use variables when providing cUrl examples to copy. 

Example:
`/rest/v1/product/{product_id}/variants`

### Steps To Reproduce

The Baserow API docs provide the following cUrl which fails to import unless the brackets are removed. 
```
curl \
-X GET \
-H "Authorization: Token YOUR_DATABASE_TOKEN" \
[https://api.baserow.io/api/database/rows/table/130952/{row_id}/?user_field_names=true](https://api.baserow.io/api/database/rows/table/130952/%7Brow_id%7D/?user_field_names=true)
```

### Public Sample App

_No response_

### Issue video log

_No response_

### Version

Cloud v1.9

### PR DESCRIPTION
**Pull request by PiyushPushkar02**
## Description

- While importing, a URI object was created to parse the query params from the URL string which doesn't allows curly brackets.
- Due to this a MalformedURLException was thrown. To avoid this removed the URI object creation.
- Added a method to split the URL string to fetch the query params

#### PR fixes following issue(s)
Fixes #19846 

#### Type of change
- Bug fix (non-breaking change which fixes an issue)

#### How Has This Been Tested?
- [x] Manual
- [x] JUnit
- [ ] Cypress

#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Test-plan-implementation#speedbreaker-features-to-consider-for-every-change) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans/_edit#areas-of-interest)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Comment by sumitsum** [comment]
Hi @PiyushPushkar02 , is the description correct ? You have mentioned that the issue is to fix the reading of query params but the query params are the ones that appear after the `?` character, and the string with the curly braces appears before, which means it is part of the URL path and not a query param e.g. here `https://api.baserow.io/api/database/rows/table/130952/{row_id}/?user_field_names=true]` the row_id part is not query param, it is part of the URL path.

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/CurlImporterServiceTest.java [hunk 1]
```diff
@@ -319,6 +319,32 @@ public void testImportActionOnEmptyInput() {
...
+
+        assert page != null;
+        Mono<ActionDTO> action = curlImporterService.importAction(urlWithCurlyBrackets, page.getId(), "actionName", workspaceId, null);
+
+        StepVerifier
+                .create(action)
+                .assertNext(actionDTO -> {
+                    assertThat(actionDTO.getDatasource().getDatasourceConfiguration().getUrl()).isNotNull();
+                    assertThat(actionDTO.getDatasource().getDatasourceConfiguration().getUrl()).isNotEmpty();
+                    assertThat(actionDTO.getActionConfiguration().getQueryParameters()).isNotEmpty();
```
Comment (sanveer-osahan) [inline app/server/appsmith-server/src/test/java/com/appsmith/server/services/CurlImporterServiceTest.java:343]: Can you add asserts that also checks for the url and params are equal to the expected ones?

**Comment by PiyushPushkar02** [comment]
Hi @sumitsum apologies for the confusion. Have updated the description for better understanding.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/CurlImporterServiceCEImpl.java [hunk 2]
```diff
@@ -492,7 +489,7 @@ private String guessTheContentType(List<String> dataParts, List<String> formPart
         return null;
     }
 
-    private void trySaveURL(ActionDTO action, String token) throws MalformedURLException, URISyntaxException {
+    private void trySaveURL(ActionDTO action, String token) throws MalformedURLException, URISyntaxException,IndexOutOfBoundsException {
```
Comment (sanveer-osahan) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/CurlImporterServiceCEImpl.java:492]: Add a space here after `,` for better code formatting.

**Review by sanveer-osahan — CHANGES_REQUESTED** [review]
It's a recommended practice to add references to the original issue in the commit messages. It will make it easy to filter out (+ cherry pick) the commits that went in your changes if suppose an issue occurs in syncing the branches.

**Review by sanveer-osahan — APPROVED** [review]

**Pull request closed without merging**